lightning/src/routing/router.rs

```diff
@@ -853,7 +861,10 @@ where L::Target: Logger {
 					let contributes_sufficient_value = available_value_contribution_msat >= minimal_value_contribution_msat;
 
 					// Do not consider candidates that exceed the maximum total cltv expiry limit.
-					let max_total_cltv_expiry_delta = payment_params.max_total_cltv_expiry_delta;
+					// We subtract 2*40 here in order to account for some of the privacy-enhancing
+					// random CLTV expiry delta offset we add on top later.
+					let max_total_cltv_expiry_delta = payment_params.max_total_cltv_expiry_delta
+						.checked_sub(2*40).unwrap_or(payment_params.max_total_cltv_expiry_delta);
```

Comment:
Is the significance of `2*40` mentioned anywhere? Could you add a constant for `40` since it's used elsewhere?

Reply:
No, it's a rough estimate of the random walk's expected value, since `40` is the most commonly used value in the network today and the random walk has a length in [1..3]. I now added an explanation and a `MEDIAN_HOP_CLTV_EXPIRY_DELTA` constant with https://github.com/lightningdevkit/rust-lightning/pull/1286/commits/4446516b4128ecc4f52d7ab991f35208fd635df6.